miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/SampleController.java

```diff
@@ -168,6 +170,23 @@ public SampleDto createSample(@RequestBody SampleDto sampleDto, UriComponentsBui
           SampleClass stockClass = sampleClassService.inferStockFromAliquot(sampleClass);
           dto.setStockClassId(stockClass.getId());
         }
+      } else if (sampleDto instanceof SampleStockDto) {
```

Comment:
Hmm good point. It should be an else, but inferring the tissue class should also happen within the aliquot block.